functests/dpdk/dpdk.go

```diff
@@ -144,135 +126,206 @@ var _ = Describe("dpdk", func() {
+
+				pod, err = client.Client.Pods(namespaces.DpdkTest).Get(context.Background(), pod.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				numberOfFreeHugePages, err := strconv.Atoi(strings.Replace(buff.String(), "\r\n", "", 1))
+				Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+
+				err = client.Client.Pods(namespaces.DpdkTest).Delete(context.Background(), pod.Name, metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64Ptr(0)})
 				Expect(err).ToNot(HaveOccurred())
 
-				return numberOfFreeHugePages
-			}, 5*time.Minute, 5*time.Second).Should(Equal(activeNumberOfFreeHugePages - 1))
+				Eventually(func() error {
+					_, err := client.Client.Pods(namespaces.DpdkTest).Get(context.Background(), pod.Name, metav1.GetOptions{})
+					if err != nil && errors.IsNotFound(err) {
+						return err
+					}
+					return nil
+				}, 10*time.Second, 1*time.Second).Should(HaveOccurred())
+			})
+		})
+	})
 
-			pod, err = client.Client.Pods(namespaces.DpdkTest).Get(context.Background(), pod.Name, metav1.GetOptions{})
+	Context("VFS split for dpdk and netdevice", func() {
+		BeforeEach(func() {
+			if discovery.Enabled() {
+				Skip("Split VF test disabled for discovery mode")
+			}
+		})
+		execute.BeforeAll(func() {
+			CleanSriov()
+			createSriovPolicyAndNetworkShared()
+			var err error
+			dpdkWorkloadPod, err = createDPDKWorkload(nodeSelector,
+				strings.ToUpper(dpdkResourceName),
+				fmt.Sprintf(SERVER_TESTPMD_COMMAND, strings.ToUpper(dpdkResourceName)),
+				60,
+				true)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
 
-			err = client.Client.Pods(namespaces.DpdkTest).Delete(context.Background(), pod.Name, metav1.DeleteOptions{GracePeriodSeconds: pointer.Int64Ptr(0)})
+			_, err = createDPDKWorkload(nodeSelector,
+				strings.ToUpper(dpdkResourceName),
+				fmt.Sprintf(CLIENT_TESTPMD_COMMAND, strings.ToUpper(dpdkResourceName)),
+				10,
+				false)
 			Expect(err).ToNot(HaveOccurred())
+		})
 
-			Eventually(func() error {
-				_, err := client.Client.Pods(namespaces.DpdkTest).Get(context.Background(), pod.Name, metav1.GetOptions{})
-				if err != nil && errors.IsNotFound(err) {
-					return err
-				}
-				return nil
-			}, 10*time.Second, 1*time.Second).Should(HaveOccurred())
+		It("should forward and receive packets from a pod running dpdk base", func() {
+			Expect(dpdkWorkloadPod).ToNot(BeNil(), "No dpdk workload pod found")
+			var out string
+			var err error
+
+			By("Parsing output from the DPDK application")
+			Eventually(func() string {
+				out, err = pods.GetLog(dpdkWorkloadPod)
+				Expect(err).ToNot(HaveOccurred())
+				return out
+			}, 8*time.Minute, 1*time.Second).Should(ContainSubstring(LOG_ENTRY),
+				"Cannot find accumulated statistics")
+			checkRxTx(out)
+		})
+
+		It("Run a regular pod using a vf shared with the dpdk's pf", func() {
+			podDefinition := pods.DefineWithNetworks(namespaces.DpdkTest, []string{"test-regular-network"})
```

Comment:
nit: can use add the network name as `const` and reuse it in the `createSriovPolicyAndNetworkShared` function?